### ISSUE TITLE
[Bug]: Layout routes on same directory level causes build failure

### ISSUE THREAD

**Issue by sjones6** [remix-run/remix#769]
### Which Remix packages are impacted?

- [X] `remix` (Remix core)
- [ ] `create-remix`
- [ ] `@remix-run/architect`
- [ ] `@remix-run/cloudflare-workers`
- [ ] `@remix-run/dev`
- [ ] `@remix-run/express`
- [ ] `@remix-run/netlify`
- [ ] `@remix-run/node`
- [ ] `@remix-run/react`
- [ ] `@remix-run/serve`
- [ ] `@remix-run/server-runtime`
- [ ] `@remix-run/vercel`

### What version of Remix are you using?

1.0.6

### Steps to Reproduce

See repro case here: https://github.com/sjones6/duplicate_template_route_repro

After running `npm i`, try `npm run dev` and you should get a build error: 

```
npm run dev

> remix-app-template@ dev /Users/spencer/Code/template_repro
> remix dev

Path undefined defined by route "routes/__foo" conflicts with route "routes/__bar"
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

### Expected Behavior

Application builds, and routes in nested template directories are registered.

### Actual Behavior

Build step errors and fails.

**Comment by abn5x** [comment]
Can confirm!

**Comment by Weyker** [comment]
fixed on #1347

**Comment by sjones6** [comment]
Closing, as this has been fixed referencing a duplicate issue.
